tests/Unit/Elliptic/Actions/Test_InitializeAnalyticSolution.cpp

```diff
@@ -25,87 +29,144 @@ struct ScalarFieldTag : db::SimpleTag {
   using type = Scalar<DataVector>;
 };
 
-template <size_t Dim>
-struct AnalyticSolution {
-  tuples::TaggedTuple<ScalarFieldTag> variables(
-      const tnsr::I<DataVector, Dim, Frame::Inertial>& x,
-      tmpl::list<ScalarFieldTag> /*meta*/) const noexcept {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
+struct AnalyticSolution;
+
+struct AnalyticSolutionOrData : PUP::able {
+  using creatable_classes =
+      tmpl::list<AnalyticSolutionOrData, AnalyticSolution>;
+
+  AnalyticSolutionOrData() = default;
+  explicit AnalyticSolutionOrData(CkMigrateMessage* m) : PUP::able(m) {}
+  // For some reason this fails to compile if replaced by just `PUPable_decl`.
+  // Upstream issue: https://github.com/UIUC-PPL/charm/issues/3235
+  WRAPPED_PUPable_decl_template(AnalyticSolutionOrData);
```

Comment:
You need to pair your `PUPable_decl` with a `PUPable_def` outside the class.

Reply:
Aha, Thanks!

(These PUPable macros have been a major headache for me.. I spent way too much time trying to figure out the correct combination of macros needed to make the compiler, linker and runtime happy..)